sched/sched/sched.h

```diff
@@ -379,7 +378,11 @@ void nxsched_sporadic_lowpriority(FAR struct tcb_s *tcb);
 void nxsched_suspend(FAR struct tcb_s *tcb);
 #endif
 
-#ifdef CONFIG_SMP
+#if defined(up_this_task)
+#  define this_task()            up_this_task()
+#elif !defined(CONFIG_SMP)
+#  define this_task()            ((FAR struct tcb_s *)g_readytorun.head)
```

Comment:
Why not use (current_task(this_cpu())) like it was before ?

Reply:
#define this_task() (current_task(this_cpu())) 
This macro can only be used in a single-core environment.